packages/common/src/hardhat/tasks/identifiers.js

```diff
@@ -24,3 +24,68 @@ task("whitelist-identifiers", "Whitelist identifiers from JSON file")
 
     await _whitelistIdentifier(web3, id, identifierWhitelist, deployer);
   });
+
+task("migrate-identifiers", "Adds all whitelisted identifiers on one IdentifierWhitelist to another")
+  .addParam("from", "The contract from which to query a whitelist of identifiers.", "", types.string)
+  .addOptionalParam("to", "The contract on which to whitelist new identifiers.", "", types.string)
+  .setAction(async function (taskArguments, hre) {
+    const { deployments, getNamedAccounts, web3 } = hre;
+    const { deployer } = await getNamedAccounts();
+    const { from, to } = taskArguments;
+
+    const IdentifierWhitelist = await deployments.get("IdentifierWhitelist");
+    const oldWhitelist = new web3.eth.Contract(IdentifierWhitelist.abi, from);
+
+    const identifiersToWhitelist = [];
+    const addedIdentifierEvents = await oldWhitelist.getPastEvents("SupportedIdentifierAdded", { fromBlock: 0 });
+
+    // Filter out identifiers that are not currently whitelisted.
+    let isIdentifierSupportedPromises = [];
+    addedIdentifierEvents.forEach((_event) => {
+      isIdentifierSupportedPromises.push(
+        oldWhitelist.methods.isIdentifierSupported(_event.returnValues.identifier).call()
+      );
+    });
+    (await Promise.all(isIdentifierSupportedPromises)).forEach((supported, i) => {
+      if (supported) {
+        identifiersToWhitelist.push(addedIdentifierEvents[i].returnValues.identifier);
+      }
+    });
```

Comment:
nit: I think this could be simplified a bit.

The first part seems more suited for a map `addedIdentifierevents.map(...` (so you don't have to declare and append).

You could also have the first map return identifier or null rather than true/false and then just filter out the nulls, which might be a bit simpler.

Reply:
done